tracing-error/src/backtrace.rs

```diff
@@ -170,8 +169,46 @@ impl fmt::Display for SpanTrace {
     }
 }
 
+struct SpanTraceFrame<'a> {
+    metadata: &'a Metadata<'a>,
+    fields: &'a str,
+}
+
+impl<'a> fmt::Debug for SpanTraceFrame<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{{ target: {:?}, name: {:?}",
+            self.metadata.target(),
+            self.metadata.name()
+        )?;
+
+        if !self.fields.is_empty() {
+            write!(f, ", fields: {:?}", self.fields)?;
+        }
+
+        if let Some((file, line)) = self
+            .metadata
+            .file()
+            .and_then(|file| self.metadata.line().map(|line| (file, line)))
+        {
+            write!(f, ", file: {:?}, line: {:?}", file, line)?;
+        }
+
+        write!(f, " }}")?;
+
+        Ok(())
+    }
+}
```

Comment:
nit/TIOLI: since this is only used in the `fmt::Debug` impl for `SpanTrace`, I might prefer to put these types inside that function?